src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java

```diff
@@ -32,16 +57,48 @@ public static boolean isValidModuleClassName(String test) { // TODO: Ensure that
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns an immutable sessions list, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public List<Session> getSessions() {
+        return Collections.unmodifiableList(sessions);
+    }
+
+    /**
+     * Returns true if both modules have the same name and session list.
+     * This defines a stronger notion of equality between two module classes.
+     *
+     * @param other the object to be compared to.
+     * @return true if objects are equal.
+     */
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof ModuleClass // instanceof handles nulls
-                && className.equals(((ModuleClass) other).className)); // state check
+                && className.equals(((ModuleClass) other).className)
+                && sessions.equals(((ModuleClass) other).sessions));
+    }
+
+    /**
+     * Returns true if both modules have the same name.
+     * This defines a weaker notion of equality between two module classes.
+     *
+     * @param otherModule the module class to be compared to.
+     * @return true if both modules have the same name.
+     */
+    public boolean isSameModuleClass(ModuleClass otherModule) {
+        return otherModule == this
+            || (otherModule != null && otherModule.className.equals(this.className));
+    }
+
+    public boolean hasSession(Session toCheck) {
+        return sessions.stream().anyMatch(toCheck::isSameSession);
     }
 
     @Override
     public int hashCode() {
-        return className.hashCode();
+        return Objects.hash(className, sessions);
```

Comment:
Just for some information, the `UniqueModuleClassList` uses a hash map to check that all classes are unique. If we are not allowing classes with the same name to exist, maybe we should either edit the `UniqueModuleClassList::isUniqueListOfModuleClasses` method to check only class name collisions, or only use the `className` for hashing this class.